kong/resty/dns/client.lua

```diff
@@ -766,6 +771,8 @@ local function executeQuery(premature, item)
   item.semaphore:post(math_max(item.semaphore:count() * -1, 1))
   item.semaphore = nil
   ngx.sleep(0)
+  -- 3) destroy the resolver -- ditto in individualQuery
+  r:destroy()
```

Comment:
Hi @robincher

Could we move `r:destroy()` after `r:query()` https://github.com/Kong/kong/pull/10691/files#diff-2001be5a8987462f9edc79d36a2f3c3a7138ae56a2aca1ff75573d19127edf7eL746.

Recently, I file a new pr to fix dns blocking problem and remove the duplicated logic via calling individualQuery(). In the individualQuery(), it will immediately release `r` via `r:destroy()` after `r:query()`.

I believe it is ok here. Could you recheck it again? ty

Reply:
@chobits Checked it out, the new individualQuery may safely assumed the socket leak not to happen.